rust-runtime/aws-smithy-http/src/byte_stream.rs

```diff
@@ -141,136 +141,145 @@ pub use bytestream_util::Length;
-///        Ok(())
-///     }
-///     ```
-///
-/// ## Getting data into a ByteStream
-/// ByteStreams can be created in one of three ways:
-/// 1. **From in-memory binary data**: ByteStreams created from in-memory data are always retryable. Data
-/// will be converted into `Bytes` enabling a cheap clone during retries.
-///     ```no_run
-///     use bytes::Bytes;
-///     use aws_smithy_http::byte_stream::ByteStream;
-///     let stream = ByteStream::from(vec![1,2,3]);
-///     let stream = ByteStream::from(Bytes::from_static(b"hello!"));
-///     ```
-///
-/// 2. **From a file**: ByteStreams created from a path can be retried. A new file descriptor will be opened if a retry occurs.
-///     ```no_run
-///     #[cfg(feature = "tokio-rt")]
-///     # {
-///     use aws_smithy_http::byte_stream::ByteStream;
-///     let stream = ByteStream::from_path("big_file.csv");
-///     # }
-///     ```
-///
-/// 3. **From an `SdkBody` directly**: For more advanced / custom use cases, a ByteStream can be created directly
-/// from an SdkBody. **When created from an SdkBody, care must be taken to ensure retriability.** An SdkBody is retryable
-/// when constructed from in-memory data or when using [`SdkBody::retryable`](crate::body::SdkBody::retryable).
-///     ```no_run
-///     use aws_smithy_http::byte_stream::ByteStream;
-///     use aws_smithy_http::body::SdkBody;
-///     use bytes::Bytes;
-///     let (mut tx, channel_body) = hyper::Body::channel();
-///     // this will not be retryable because the SDK has no way to replay this stream
-///     let stream = ByteStream::new(SdkBody::from(channel_body));
-///     tx.send_data(Bytes::from_static(b"hello world!"));
-///     tx.send_data(Bytes::from_static(b"hello again!"));
-///     // NOTE! You must ensure that `tx` is dropped to ensure that EOF is sent
-///     ```
-///
-#[pin_project]
-#[derive(Debug)]
-pub struct ByteStream(#[pin] Inner<SdkBody>);
+// pin_project_lite can't handle triple-slash doc comments
+pin_project! {
+    #[doc = "Stream of binary data"]
+    #[doc = ""]
+    #[doc = "`ByteStream` wraps a stream of binary data for ease of use."]
+    #[doc = ""]
+    #[doc = "## Getting data out of a `ByteStream`"]
+    #[doc = ""]
+    #[doc = "`ByteStream` provides two primary mechanisms for accessing the data:"]
+    #[doc = "1. With `.collect()`:"]
+    #[doc = ""]
+    #[doc = "    [`.collect()`](crate::byte_stream::ByteStream::collect) reads the complete ByteStream into memory and stores it in `AggregatedBytes`,"]
+    #[doc = "    a non-contiguous ByteBuffer."]
+    #[doc = "    ```no_run"]
+    #[doc = "    use aws_smithy_http::byte_stream::{ByteStream, AggregatedBytes};"]
+    #[doc = "    use aws_smithy_http::body::SdkBody;"]
+    #[doc = "    use bytes::Buf;"]
+    #[doc = "    async fn example() {"]
+    #[doc = "       let stream = ByteStream::new(SdkBody::from(\"hello! This is some data\"));"]
+    #[doc = "       // Load data from the stream into memory:"]
```

Comment:
hmm, are you sure this is required here? There are other places where we use pin_project_lite and doc comments:

https://github.com/awslabs/smithy-rs/blob/a0539e20b069a7de021c84521d8f3c7ba098ad6d/rust-runtime/aws-smithy-async/src/future/fn_stream.rs#L16